reverse_engineering/helpers/sequenceHelper.js

```diff
@@ -0,0 +1,290 @@
+  }
+};
+/**
+ * @param {{ execute: Execute }}
+ * @returns {({ schema, logger }: { schema: string, logger: object }) => Promise<Sequence[]>}
+ */
+const getSchemaSequences =
+  ({ execute }) =>
+  async ({ schema, logger }) => {
+    const sequences = await getSchemaSequenceDtos({ schema, execute, logger });
+
+    return sequences.map((sequenceData) => {
+      const sharing = getSequenceSharing({ sequenceData });
+      const options = getBooleanOptions({ sequenceData });
+      const optionsFromDdl = getOptionsFromDdl({ sequenceData });
+      const extendOptions = getExtendOptions({ sequenceData });
+      const cache = sequenceData.cacheValue
+        ? SEQUENCE_OPTION.cache
+        : SEQUENCE_OPTION.noCache;
+
+      return {
+        sequenceName: sequenceData.sequenceName,
+        minValue: sequenceData.minValue,
+        maxValue: sequenceData.maxValue,
+        increment: sequenceData.increment,
+        cacheValue: sequenceData.cacheValue,
+        cache,
+        sharing,
+        ...options,
+        ...optionsFromDdl,
+        ...extendOptions,
+      };
+    });
+  };
+
+/**
+ * @param {{ sequenceData: SequenceDto }}
+ * @returns {string}
+ */
+const getSequenceSharing = ({ sequenceData }) => {
+  return sequenceData.sharing?.replace(SHARING_LINK_REGEX, '')?.toLowerCase();
+};
+
+/**
+ * @param {{ sequenceData: SequenceDto }}
+ * @returns {Options}
+ */
+const getExtendOptions = ({ sequenceData }) => {
+  const scaleValue = sequenceData[SEQUENCE_OPTION.scale];
+  const shardValue = sequenceData[SEQUENCE_OPTION.shard];
+  const extendValue = sequenceData[SEQUENCE_OPTION.extend];
+  const extend = SEQUENCE_OPTION_MAP[SEQUENCE_OPTION.extend][extendValue];
+  const scaleExtend = scaleValue === OPTION_VALUE.true ? extend : '';
+  const shardExtend = shardValue === OPTION_VALUE.true ? extend : '';
+
+  return {
+    scaleExtend,
+    shardExtend,
+  };
+};
+
+/**
+ * @param {{ sequenceData: SequenceDto }}
+ * @returns {Options}
+ */
+const getBooleanOptions = ({ sequenceData }) => {
+  return Object.entries(SEQUENCE_OPTION_MAP).reduce(
+    (result, [optionName, optionMap]) => {
+      return {
+        ...result,
+        [optionName]: optionMap[sequenceData[optionName]],
+      };
+    },
+    {}
+  );
+};
+
+/**
+ * @param {{ sequenceData }}
+ * @returns {Options}
+ */
+const getOptionsFromDdl = ({ sequenceData }) => {
+  const start = getStartOptionFromDdl({ sequenceData });
+
+  return {
+    start,
+  };
+};
+
+/**
+ * @param {{ sequenceData }}
+ * @returns {number | undefined}
+ */
+const getStartOptionFromDdl = ({ sequenceData }) => {
+  const ddlScript = sequenceData.ddlScript;
+
+  if (!ddlScript.includes('START WITH')) {
+    return sequenceData.minValue < sequenceData.maxValue
+      ? sequenceData.minValue
+      : sequenceData.maxValue;
+  }
+
+  const start = ddlScript.match(START_VALUE_REGEX)?.[2];
+
+  return !isNaN(Number(start)) ? Number(start) : start;
+};
+
+module.exports = {
+  getSchemaSequences,
```

Comment:
if we expect the main module export to be a single method, let's give module a name of that function 🙂 no more ambiguous helpers